### PR TITLE
Fix carousel alignment and test perf improvement

### LINKED ISSUE #1635: Investigate alignment on carousel
> Might broken by #1560. :(

Type `carousel`.

image

### PR DESCRIPTION
**Pull request by compulim**
> Fix #1635.

## Background

Alignment was broken due to #1560 and #1561 when showing carousel with bot avatar initials.

This PR will also add tests for carousel and stacked layout, and a bit more stuff:

- Test perf improvement for suggested actions tests
   - Replaced sleep with [conditions](https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_Condition.html) (a.k.a. proactive retry)
   - Updated snapshots with cursor removed
- Added first page object: `hideCursor`
   - Will evaluate how effective this coding pattern is, and add more later
- Added timeout settings via `constants.json`

All tests running locally on my box took 49 seconds.

### Fixed
- `component`: Fix [#1560](https://github.com/Microsoft/BotFramework-WebChat/issues/1560) and [#1635](https://github.com/Microsoft/BotFramework-WebChat/issues/1635). Fixed carousel layout did not show date and alignment issues, by [@compulim](https://github.com/compulim) in PR [#1561](https://github.com/Microsoft/BotFramework-WebChat/pull/1561) and [#1641](https://github.com/Microsoft/BotFramework-WebChat/pull/1641)

### CODE REVIEW

**Comment by corinagum** [comment]
Partial fix for https://github.com/Microsoft/BotFramework-WebChat/issues/1625

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   - `connectCarouselFilmStrip`: Fixed `botAvatarInitials` and `userAvatarInitials` functionality from [recent name change](https://github.com/Microsoft/BotFramework-WebChat/pull/1486),
   - `BasicTranscript`: Fixed user activity should not be recreated after receive ACK from Direct Line,
   - by [@corinagum](https://github.com/corinagum) in PR [#1528](https://github.com/Microsoft/BotFramework-WebChat/pull/1528)
-- `component`: Fix [#1560](https://github.com/Microsoft/BotFramework-WebChat/issues/1560). Fixed carousel layout did not show date and alignment issues, by [@compulim](https://github.com/compulim) in PR [#1561](https://github.com/Microsoft/BotFramework-WebChat/pull/1561)
+- `component`: Fix [#1560](https://github.com/Microsoft/BotFramework-WebChat/issues/1560) and [#1635](https://github.com/Microsoft/BotFramework-WebChat/issues/1635). Fixed carousel layout did not show date and alignment issues, by [@compulim](https://github.com/compulim) in PR [#1561](https://github.com/Microsoft/BotFramework-WebChat/pull/1561) and [#1641](https://github.com/Microsoft/BotFramework-WebChat/pull/1641)
```
[the remark this answers is not in the document]
Comment (compulim) [reply CHANGELOG.md]: Fixed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   - `connectCarouselFilmStrip`: Fixed `botAvatarInitials` and `userAvatarInitials` functionality from [recent name change](https://github.com/Microsoft/BotFramework-WebChat/pull/1486),
   - `BasicTranscript`: Fixed user activity should not be recreated after receive ACK from Direct Line,
   - by [@corinagum](https://github.com/corinagum) in PR [#1528](https://github.com/Microsoft/BotFramework-WebChat/pull/1528)
-- `component`: Fix [#1560](https://github.com/Microsoft/BotFramework-WebChat/issues/1560). Fixed carousel layout did not show date and alignment issues, by [@compulim](https://github.com/compulim) in PR [#1561](https://github.com/Microsoft/BotFramework-WebChat/pull/1561)
+- `component`: Fix [#1560](https://github.com/Microsoft/BotFramework-WebChat/issues/1560) and [#1635](https://github.com/Microsoft/BotFramework-WebChat/issues/1635). Fixed carousel layout not showing date and alignment issues, by [@compulim](https://github.com/compulim) in PR [#1561](https://github.com/Microsoft/BotFramework-WebChat/pull/1561) and [#1641](https://github.com/Microsoft/BotFramework-WebChat/pull/1641)
```
Comment (corinagum) [inline CHANGELOG.md]: Add issue #1625

File: __tests__/basic.js [hunk 3]
```diff
@@ -9,21 +11,19 @@ import webChatLoaded from './setup/conditions/webChatLoaded';
...
-  const { driver } = await setupWebDriver();
+  const { driver, pageObjects } = await setupWebDriver();
 
-  await driver.wait(webChatLoaded(), 2000);
-  await driver.wait(directLineConnected(), 2000);
+  await driver.wait(webChatLoaded(), timeouts.navigation);
+  await driver.wait(directLineConnected(), timeouts.directLine);
 
-  const input = await driver.findElement(By.tagName('input[type="text"]'));
+  const input = await driver.findElement(By.css('input[type="text"]'));
```
Comment (corinagum) [inline __tests__/basic.js]: question: Why do we want to use `css` instead of `tagName`?

Comment (compulim) [reply __tests__/basic.js]: `tagName` should be just `input`, it does not support attribute `role`.

In contrast, `css` is the CSS selector, which could be `input[type="text"]`.

File: CHANGELOG.md (hunk 2 above)
Comment (compulim) [reply CHANGELOG.md]: Added.

**Pull request merged**